test_scripts/API/VehicleData/GearStatus/common.lua

```diff
@@ -0,0 +1,441 @@
+    RAISE_EVENT(event, event)
+  end)
+  :Timeout(timeout)
+  local function forceStopSDL()
+    if isSDLShutDownSuccessfully == false then
+      utils.cprint(35, "SDL was shutdown forcibly")
+      RAISE_EVENT(event, event)
+    end
+  end
+  RUN_AFTER(forceStopSDL, timeout + 500)
+end
+
+--[[ @unexpectedDisconnect: closing connection
+--! @parameters: none
+--! @return: none
+--]]
+function m.unexpectedDisconnect()
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(actions.mobile.getAppsCount())
+  actions.mobile.disconnect()
+  utils.wait(1000)
+end
+
+--[[ @checkResumption_FULL: function that checks HMI level resumption to FULL level
+--! @parameters: none
+--! @return: none
+--]]
+function m.checkResumption_FULL()
+  m.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp", {})
+  :Do(function(_, data)
+    m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  m.getMobileSession():ExpectNotification("OnHMIStatus",
+    { hmiLevel = "NONE" },
+    { hmiLevel = "FULL" })
+  :Times(2)
+end
+
+--[[ @checkResumption_NONE: function that checks HMI level resumption to NONE level
+--! @parameters: none
+--! @return: none
+--]]
+function m.checkResumption_NONE()
+  m.getMobileSession():ExpectNotification("OnHMIStatus",{ hmiLevel = "NONE" })
+end
+
+--[[ @registerAppWithResumption: Successful app registration with resumption
+--! @parameters:
+--! pAppId: application number (1, 2, etc.)
+--! pLevelCheckFunc: function for expectation of HMI level related messages
+--! isHMIsubscription: if true VD.SubscribeVehicleData request is expected on HMI, otherwise - not expected
+--! @return: none
+--]]
+function m.registerAppWithResumption(pAppId, pLevelCheckFunc, isHMIsubscription)
+  if not pAppId then pAppId = 1 end
+  m.getMobileSession(pAppId):StartService(7)
+  :Do(function()
+    m.getConfigAppParams(pAppId).hashID = m.getHashId(pAppId)
+    local corId = m.getMobileSession(pAppId):SendRPC("RegisterAppInterface", m.getConfigAppParams(pAppId))
+    m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+      application = { appName = m.getConfigAppParams(pAppId).appName }
+    })
+    :Do(function(_, data)
+      if true == isHMIsubscription then
+        m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { gearStatus = true })
+        m.getHMIConnection():SendResponse( data.id, data.method, "SUCCESS", { gearStatus = m.subUnsubResponse } )
+      else
+        m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData"):Times(0)
+      end
+    end)
+    m.getMobileSession(pAppId):ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+    :Do(function()
+      m.getMobileSession(pAppId):ExpectNotification("OnPermissionsChange")
+    end)
+  end)
+  pLevelCheckFunc(pAppId)
+end
+
+--[[ @postcondition: Stop SDL and restore sdl_preloaded_pt.json file
```

Comment:
Please remove custom postcondition function, because the function for preloaded file restore is called in precondition function from actions.